Loadout Revamp

FearTheBlackout opened the pull request:
:cl: Devildabeast
tweak: Peacekeeper armbands are now restricted to Fleet instead of SolGov roles.
tweak: SolGov awards can now be worn by a select few civilians (SCGR & FT)
tweak: NanoTrasen medals have now been changed to corporate medals, and will be named after your character's faction.
tweak: Corporate medals can now be worn by all civilians.
tweak: EC caps are no longer restricted to SolGov roles (only the EC branch).
rscadd: The corporate merit medal has been added to the loadout.
/:cl:

The only outstanding problem I can think of is that civilian FTs will technically be able to wear SolGov medals, but I somehow don't see that as being a big problem.

Demonstration of corporate medals below:
![medals](http://puu.sh/B0OFB/7a22ac98a6.png)

babydoll commented on maps/torch/loadout/loadout_accessories.dm:
```diff
@@ -23,7 +24,9 @@
 	cost = 5
-	allowed_branches = UNIFORMED_BRANCHES
+	allowed_branches = list(/datum/mil_branch/expeditionary_corps, /datum/mil_branch/fleet, /datum/mil_branch/civilian)
+	allowed_roles = list(/datum/job/captain, /datum/job/hop, /datum/job/cmo, /datum/job/chief_engineer, /datum/job/hos, /datum/job/bridgeofficer, /datum/job/sea, /datum/job/senior_engineer, /datum/job/engineer, /datum/job/warden, /datum/job/detective, /datum/job/officer, /datum/job/senior_doctor, /datum/job/doctor, /datum/job/qm, /datum/job/cargo_tech, /datum/job/janitor, /datum/job/chef, /datum/job ...
```
no

FearTheBlackout replied:
Why not?

babydoll replied:
This is impossible to parse. What is missing/added from one of the defines?

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+	if(H.personal_faction == "None")
+		name = "corporate command medal"
+		desc = "A gold medal awarded to corporate employees for service as the Captain of a corporate facility, station, or vessel."
+	else return
```
Superfluous else and return

afterthought2 replied:
Trailing return. Also not a fan of the general implementation.

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+	H = get_holder_of_type(src, /mob/living/carbon/human)
+	if(H)
+		set_desc(H)
+	if(H.personal_faction == "NanoTrasen")
```
This will runtime if H is null

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+		set_desc(H)
+	if(H.personal_faction == "NanoTrasen")
+		icon_state = "gold_nt"
+	if(H.personal_faction == "None")
```
This will runtime if H is null

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+	if(H.personal_faction == "None")
+		name = "corporate service medal"
+		desc = "A silver medal awarded to corporate employees for distinguished service in support of corporate interests."
+	else return
```
Same as above.

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+
+/obj/item/clothing/accessory/medal/silver/corporate
+
+/obj/item/clothing/accessory/medal/silver/corporate/Initialize()
```
This looks like copy-paste of the gold version with minor tweaks

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
-	name = "\improper NanoTrasen command medal"
-	desc = "A gold medal awarded to NanoTrasen employees for service as the Captain of a NanoTrasen facility, station, or vessel."
-	icon_state = "gold_nt"
+/obj/item/clothing/accessory/medal/gold/corporate
```
There is no need to declare empty type definitions

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+	name = "\improper [H.personal_faction] command medal"
+	desc = "A gold medal awarded to [H.personal_faction] employees for service as the Captain of a [H.personal_faction] facility, station, or vessel."
+
+/obj/item/clothing/accessory/medal/silver/corporate
```
There is no need to declare empty type definitions

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+/obj/item/clothing/accessory/medal/bronze/corporate
 
-/obj/item/clothing/accessory/medal/iron/nanotrasen
+/obj/item/clothing/accessory/medal/bronze/corporate/Initialize()
```
This looks like copy-paste of the gold/silver versions, with minor tweaks

babydoll commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -6,6 +6,8 @@
 	icon_state = "bronze"
 	slot = ACCESSORY_SLOT_MEDAL
 
+/obj/item/clothing/accessory/medal/proc/set_desc(var/mob/living/carbon/human/H)
```
Why not put the actual content of the proc here and some vars to change the 1/2 words you want changed per individual medal? Saves having to overwrite this blank proc each time.

"A [material_desc] medal awarded to [awardee_desc] for [award_desc] of a [h,personal_faction] ship, whatever"

PsiOmegaDelta commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
\ No newline at end of file
+	icon_state = "iron_nt"
+
+/obj/item/clothing/accessory/medal/iron/corporate/Initialize()
```
This looks like copy-paste of the gold/silver/bronze versions with minor tweaks

afterthought2 commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+/obj/item/clothing/accessory/medal/silver/corporate/set_desc(var/mob/living/carbon/human/H)
+	if(!istype(H))
+		return
+	name = "\improper [H.personal_faction] service medal"
```
Do not change the name without using `SetName`.

afterthought2 commented on code/modules/clothing/under/accessories/medals.dm:
```diff
@@ -32,22 +34,89 @@
+	if(H.personal_faction == "None")
+		name = "\improper corporate service medal"
+		desc = "A silver medal awarded to corporate employees for distinguished service in support of corporate interests."
+	else return
```
Lots of copypaste. Use subtypes, store things that change between types in vars, pull out repeated calls into shared procs.

Commit:
Accessory Revamp

:cl: Devildabeast
tweak: Peacekeeper armbands are now restricted to Fleet instead of SolGov roles.
tweak: SolGov awards can now be worn by a select few civilians (SCGR & FT)
tweak: NanoTrasen medals have now been changed to corporate medals, and will be named after your character's faction.
tweak: Corporate medals can now be worn by all civilians.
/:cl:

The only outstanding problem I can think of is that civilian FTs will technically be able to wear SolGov medals, but I somehow don't see that as being a big problem.

Demonstration of corporate medals below:

OhEmGee commented:
What utility do peacekeeper armbands even have?

ghost commented:
I don't think they have any lore to them, but I like to imagine they're the accessory equivalent of screaming "I'M SPECIAL FORCES!!!" in the US Army -- but don't quote me on that.

Going to ask for some help on #coding in a while later addressing the reviews above as I have no idea what the vast majority of them mean.

comma commented:
>What utility do peacekeeper armbands even have?

None. If contents of e-armory is any indication, they go with a set to peacekeeper helmets and vests. IMO should be just a box of them in e-armory for drafted losers who didn't get a vest.
